admin/src/main/java/edu/unc/lib/dl/admin/controller/ResubmitIngestController.java

```diff
@@ -0,0 +1,176 @@
+package edu.unc.lib.dl.admin.controller;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+import java.util.HashMap;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import edu.unc.lib.dl.acl.util.AccessGroupConstants;
+import edu.unc.lib.dl.acl.util.AccessGroupSet;
+import edu.unc.lib.dl.acl.util.GroupsThreadStore;
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositField;
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositAction;
+import edu.unc.lib.dl.util.RedisWorkerConstants.DepositState;
+import edu.unc.lib.dl.util.MetsHeaderScanner;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.util.DepositConstants;
+import edu.unc.lib.dl.util.DepositStatusFactory;
+
+@Controller
+public class ResubmitIngestController {
+	
+	private static final Logger log = LoggerFactory.getLogger(ResubmitIngestController.class);
+	
+	@Autowired
+	private DepositStatusFactory depositStatusFactory;
+	
+	@Autowired
+	private File depositsDirectory;
+	
+	@RequestMapping(value = "ingest/{pid}/resubmit", method = RequestMethod.POST)
+	public @ResponseBody
+	Map<String, ? extends Object> handleResubmit(@PathVariable("pid") String pid, @RequestParam("file") MultipartFile uploadedFile, HttpServletRequest request, HttpServletResponse response) {
+		
+		Map<String, Object> result = new HashMap<String, Object>();
+		PID depositPid = new PID(pid);
+		
+		
+		// Try to retrieve the deposit status
+		
+		Map<String, String> status = depositStatusFactory.get(depositPid.getUUID());
+		
+		if (status == null || status.isEmpty()) {
+			response.setStatus(400);
+			result.put("error", "Couldn't find the deposit.");
+			return result;
+		}
+		
+		
+		// Check that the current user is either an admin or was the depositor
+		
+		AccessGroupSet groups = GroupsThreadStore.getGroups();
+		String depositorName = status.get(DepositField.depositorName.name());
+		
+		if (!(groups.contains(AccessGroupConstants.ADMIN_GROUP) || depositorName.equals(GroupsThreadStore.getUsername()))) {
+			response.setStatus(401);
+			return result;
+		}
+		
+		
+		// Check that the deposit is in the failed state
+		
+		String state = status.get(DepositField.state.name());
+		
+		if (state == null || !state.equals(DepositState.failed.name())) {
```

Comment:
Perhaps it should allow you to resubmit if the ingest is paused?  Mainly we want to prevent resubmission when an ingest is active or could become active at any moment.